sm/sm.h

```diff
@@ -708,9 +708,6 @@ M0_INTERNAL bool m0_sm_timeout_is_armed(const struct m0_sm_timeout *to);
 
 /**
  * Posts an AST to a group.
- *
```

Comment:
m0_sm_ast_post() is not only used to wake up a FOM. It is used very widely in Motr.
This statement is true for all the use cases.
I don't think we can remove this requirement.

Reply:
We can remove it now. I don't see why not. I've added more explanations there, please check the last commit.

Reply:
m0_fom_wakeup() can be called multiple times and can be called concurrent.
But the m0_sm_ast_post() still has this requirement.

Reply:
`m0_sm_ast_post()` also can be called multiple times and concurrent, but with a condition. See the latest doc.

Reply:
No more comments from me.

Reply:
Thanks @huanghua78. Do you approve it then?